src/mw/device/sort_archetype.cpp

```diff
@@ -1208,7 +1209,7 @@ void SortArchetypeNodeBase::resizeTable(int32_t)
 {
     mwGPU::getStateManager()->resizeArchetype(
         archetypeID, bins[(numPasses - 1) * 256 + 255]);
-    numDynamicInvocations = numRows;
+    numDynamicInvocations = bins[(numPasses - 1) * 256 + 255];
```

Comment:
Can you read bins once in this function and use both to resize + set numDynamicInvocations?